utilities/PerformanceTesting/OpenClosePortsLoad/open-close-ports-load.py

```diff
@@ -0,0 +1,49 @@
+import argparse
+from argparse import RawTextHelpFormatter
+import os, sys
+import random
+import time
+import subprocess
+
+
+def main(num_ports, num_per_second):
+    ports = [False] * (num_ports + 1)
+    procs = [None] * (num_ports + 1)
+    while True:
+        time.sleep(1.0 / float(num_per_second))
+        rand_port = random.randint(1, num_ports)
+        print("rand_port", rand_port)
+        if not ports[rand_port]: 
+            try:
+                ports[rand_port] = True
+                procs[rand_port] = subprocess.Popen(["socat", "TCP-LISTEN:" + str(rand_port) + ",fork", "STDOUT"])
+            except (subprocess.CalledProcessError, BlockingIOError) as e:
+                print(e)
+                ports[rand_port] = False
+        else:
+            try:
+                ports[rand_port] = False
+                procs[rand_port].terminate()
+            except (subprocess.CalledProcessError, BlockingIOError) as e:
+                print(e)
+                ports[rand_port] = True
```

Comment:
I think we can simplify this to a single list by setting `procs[rand_port]` to `None` after we terminate it
```suggestion
    procs = [None] * (num_ports + 1)
    while True:
        time.sleep(1.0 / float(num_per_second))
        rand_port = random.randint(1, num_ports)
        print("rand_port", rand_port)
        if procs[rand_port] is None: 
            try:
                procs[rand_port] = subprocess.Popen(["socat", "TCP-LISTEN:" + str(rand_port) + ",fork", "STDOUT"])
            except (subprocess.CalledProcessError, BlockingIOError) as e:
                print(e)
        else:
            try:
                procs[rand_port].terminate()
            except (subprocess.CalledProcessError, BlockingIOError) as e:
                print(e)
            finally:
                procs[rand_port] = None
```